ext/geometry/lib.rs

```diff
@@ -0,0 +1,3067 @@
+      scope,
+      value,
+      "Failed to execute 'DOMMatrixReadOnly.fromFloat32Array'".into(),
+      (|| Cow::Borrowed("Argument 1")).into(),
+      &Default::default(),
+    )?;
+    let float64 = float64.into_iter().map(|f| *f).collect::<Vec<f64>>();
+
+    let ro = if let [a, b, c, d, e, f] = float64.as_slice() {
+      DOMMatrixReadOnly {
+        #[rustfmt::skip]
+        inner: RefCell::new(Matrix4::new(
+           *a,  *c, 0.0,  *e,
+           *b,  *d, 0.0,  *f,
+          0.0, 0.0, 1.0, 0.0,
+          0.0, 0.0, 0.0, 1.0,
+        )),
+        is_2d: Cell::new(true),
+      }
+    } else if float64.len() == 16 {
+      DOMMatrixReadOnly {
+        inner: RefCell::new(Matrix4::from_column_slice(float64.as_slice())),
+        is_2d: Cell::new(false),
+      }
+    } else {
+      return Err(GeometryError::InvalidSequenceSize);
+    };
+
+    let obj = cppgc::make_cppgc_empty_object::<DOMMatrix>(scope);
+    Ok(cppgc::wrap_object2(scope, obj, (ro, DOMMatrix {})))
+  }
+
+  #[rename("fromFloat64Array")]
+  #[required(1)]
+  #[static_method]
+  fn from_float64_array<'a>(
+    scope: &mut v8::HandleScope<'a>,
+    value: v8::Local<'a, v8::Value>,
+  ) -> Result<v8::Local<'a, v8::Object>, GeometryError> {
+    if !value.is_float64_array() {
+      return Err(GeometryError::TypeMismatch);
+    }
+    let float64 = Vec::<webidl::UnrestrictedDouble>::convert(
+      scope,
+      value,
+      "Failed to execute 'DOMMatrixReadOnly.fromFloat64Array'".into(),
+      (|| Cow::Borrowed("Argument 1")).into(),
+      &Default::default(),
+    )?;
+    let float64 = float64.into_iter().map(|f| *f).collect::<Vec<f64>>();
+
+    let ro = if let [a, b, c, d, e, f] = float64.as_slice() {
+      DOMMatrixReadOnly {
+        #[rustfmt::skip]
+        inner: RefCell::new(Matrix4::new(
+           *a,  *c, 0.0,  *e,
+           *b,  *d, 0.0,  *f,
+          0.0, 0.0, 1.0, 0.0,
+          0.0, 0.0, 0.0, 1.0,
+        )),
+        is_2d: Cell::new(true),
+      }
+    } else if float64.len() == 16 {
+      DOMMatrixReadOnly {
+        inner: RefCell::new(Matrix4::from_column_slice(float64.as_slice())),
+        is_2d: Cell::new(false),
+      }
+    } else {
+      return Err(GeometryError::InvalidSequenceSize);
+    };
+
+    let obj = cppgc::make_cppgc_empty_object::<DOMMatrix>(scope);
+    Ok(cppgc::wrap_object2(scope, obj, (ro, DOMMatrix {})))
+  }
+
+  #[fast]
+  #[getter]
+  fn a(&self, #[proto] ro: &DOMMatrixReadOnly) -> f64 {
+    ro.a_inner()
+  }
+
+  #[setter]
+  fn a(
+    &self,
+    #[webidl] value: webidl::UnrestrictedDouble,
+    #[proto] ro: &DOMMatrixReadOnly,
+  ) {
+    ro.inner.borrow_mut()[INDEX_A] = *value;
+  }
+
+  #[fast]
+  #[getter]
+  fn b(&self, #[proto] ro: &DOMMatrixReadOnly) -> f64 {
+    ro.b_inner()
+  }
+
+  #[setter]
+  fn b(
+    &self,
+    #[webidl] value: webidl::UnrestrictedDouble,
+    #[proto] ro: &DOMMatrixReadOnly,
+  ) {
+    // SAFETY: in-range access
```

Comment:
The safety comments are now unnecessary.

Reply:
thx!